docs/design/restic.rst

```diff
@@ -0,0 +1,158 @@
+=======================
+Restic-based data mover
+=======================
+
+.. admonition:: Enhancement status
+
+   Status: Proposed
+
+This is a proposal to add `Restic <https://restic.readthedocs.io/en/stable/>`_
+as an additional data mover within Scribe. Restic is a data backup utility that
+copies the data to an object store (among other options).
+
+While the main purpose of Scribe is to perform asynchronous data replication,
+there are some use cases that are more "backup oriented" but that don't require
+a full backup application (such as Velero). For example, some users may deploy
+and version control their application via GitOps techniques. These users may be
+looking for a simple method that allows preserving (off-cluster) snapshots of
+their storage so that it can be restored if necessary.
+
+Considerations
+==============
+
+The ReplicationSource and ReplicationDestination CRs of Scribe would correspond
+to the ``backup`` and ``restore`` operations, respectively, of Restic.
+Furthermore, there are repository maintenance operations that need to be
+addressed. For example, Restic manages the retention of old backups (via its
+``forget`` operation) as well as freeing objects that are no longer used (via
+its ``prune`` operation).
+
+While both Restic and Rclone read/write to object storage, their strengths are
+significantly different. The Rclone data mover is primarily designed for
+managing 1-to-many replication relationships, using the object store as an
```

Comment:
Would we be planning to implement restic in a 1:many way?

Reply:
I don't think there will be anything stopping the user from creating multiple RDs, effectively creating a 1:many.

There may be contention issues on repo locks (to be investigated). The biggest issue is that restic doesn't support "incremental" restores (i.e., only restore the files that are out of sync), so using it for data synchronization may not be very useful.